src/extractor/scripting_environment_lua.cpp

```diff
@@ -598,6 +615,9 @@ void Sol2ScriptingEnvironment::ProcessTurn(ExtractionTurn &turn)
         {
             turn_function(turn);
 
+            if( turn.weight == MAX_TURN_WEIGHT )
+              turn.weight = context.properties.GetMaxTurnWeight();
```

Comment:
the only purpose of the turn penalty maximum weight is to prevent int32_t overflows on c++ side. If turn penalties are modified on c++ then side the maximum value is not needed in Lua, but number of modifications must be reported as a warning.